packages/core/admin/admin/src/content-manager/components/ListViewTable/BulkActionButtons/index.js

```diff
@@ -225,59 +389,64 @@ const BulkActionButtons = ({
     }
   };
 
-  const handleBulkAction = async (confirmAction, toggleModal) => {
+  const handleBulkAction = async (confirmAction, toggleDialog) => {
     try {
       setIsConfirmButtonLoading(true);
       await confirmAction(selectedEntries);
       setIsConfirmButtonLoading(false);
-      toggleModal();
+      toggleDialog();
       setSelectedEntries([]);
     } catch (error) {
       setIsConfirmButtonLoading(false);
-      toggleModal();
+      toggleDialog();
     }
   };
 
-  const handleBulkDelete = () => handleBulkAction(onConfirmDeleteAll, toggleDeleteModal);
-  const handleBulkPublish = () => handleBulkAction(onConfirmPublishAll, togglePublishModal);
-  const handleBulkUnpublish = () => handleBulkAction(onConfirmUnpublishAll, toggleUnpublishModal);
+  const handleBulkDelete = () => handleBulkAction(onConfirmDeleteAll, toggleDeleteDialog);
+  const handleBulkUnpublish = () => handleBulkAction(onConfirmUnpublishAll, toggleUnpublishDialog);
+  const handleBulkPublish = () => handleBulkAction(onConfirmPublishAll, togglePublishDialog);
 
   return (
     <>
       {publishButtonIsShown && (
         <>
-          <Button variant="tertiary" onClick={togglePublishModal}>
+          <Button variant="tertiary" onClick={() => setIsSelectedEntriesModalOpen(true)}>
             {formatMessage({ id: 'app.utils.publish', defaultMessage: 'Publish' })}
           </Button>
+          <SelectedEntriesModal
+            isOpen={isSelectedEntriesModalOpen}
+            onToggle={() => setIsSelectedEntriesModalOpen((prev) => !prev)}
```

Comment:
maybe you can create and handler just for that

Reply:
do you mean extracting that function to a variable?

Reply:
yes